core/driver/sources/src/main/scala/org/virtuslab/ideprobe/wait/WaitingWithEnsurePeriod.scala

```diff
@@ -0,0 +1,71 @@
+package org.virtuslab.ideprobe
+package wait
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+class WaitingWithEnsurePeriod(
```

Comment:
This one is very complex, I think it needs some documentation. Apart from this, IMO we could rename it to something like: `WaitingUntilConditionIsStable`

Reply:
done